src/AuthenticationService.php

```diff
@@ -220,21 +220,18 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
      * Clears the identity from authenticators that store them and the request
      *
      * @param \Psr\Http\Message\ServerRequestInterface $request The request.
+     * @param \Psr\Http\Message\ResponseInterface The response.
      * @return \Psr\Http\Message\ServerRequestInterface
      */
-    public function clearIdentity(ServerRequestInterface $request, $requestAttribute = 'identity')
+    public function clearIdentity(ServerRequestInterface $request, ResponseInterface $response)
     {
         foreach ($this->_authenticators as $authenticator) {
             if ($authenticator instanceof PersistenceInterface) {
                 $authenticator->clearIdentity($request);
```

Comment:
If the authenticators need to remove a cookie, or add a header, they'd need to return a response as the parameter would be immutable.